spec/controllers/case_management/tax_returns_controller_spec.rb

```diff
@@ -100,55 +100,260 @@
+            get :edit_status, params: params
+
+            expect(response.body).to have_text "This client requested French for their interview"
+          end
+        end
       end
     end
   end
 
-  describe "#edit_status" do
-    let(:user) { create :user_with_org }
-    let(:tax_return) { create :tax_return, client: create(:client, vita_partner: user.vita_partner) }
-    let(:params) { { id: tax_return.id, client_id: tax_return.client } }
+  describe "#update_status" do
+    let(:user) { create :user, vita_partner: (create :vita_partner) }
+    let(:client) { create :client, vita_partner: user.vita_partner }
+    let!(:intake) { create :intake, email_address: "gob@example.com", sms_phone_number: "+14155551212", client: client }
+    let(:tax_return) { create :tax_return, status: "intake_in_progress", client: client }
+    let(:status) { "review_complete_signature_requested"}
+    let(:locale) { "en" }
+    let(:internal_note) { "" }
+    let(:message_body) { "" }
+    let(:contact_method) { "email" }
+    let(:params) do
+      {
+        client_id: tax_return.client,
+        id: tax_return.id,
+        case_management_take_action_form: {
+          status: status,
+          internal_note: internal_note,
+          locale: locale,
+          message_body: message_body,
+          contact_method: contact_method,
+        }
+      }
+    end
 
-    it_behaves_like :a_get_action_for_authenticated_users_only, action: :edit_status
+    it_behaves_like :a_post_action_for_authenticated_users_only, action: :update_status
 
     context "as an authenticated user" do
       before { sign_in user }
 
-      it "returns an ok response" do
-        post :edit_status, params: params
-        expect(response).to be_ok
+      it "redirects to the messages tab with a basic flash success message" do
+        post :update_status, params: params
+
+        expect(response).to redirect_to case_management_client_path(id: tax_return.client)
+        expect(flash[:notice]).to eq "Success: Action taken! Updated status."
       end
 
-      it "finds the tax return" do
-        get :edit_status, params: params
+      context "when a new status is submitted" do
+        let(:status) { "intake_needs_assignment" }
 
-        expect(assigns(:tax_return)).to eq(tax_return)
+        it "creates a system note and updates the status" do
+          expect(SystemNote).to receive(:create_status_change_note).with(user, tax_return)
+
+          post :update_status, params: params
+          expect(tax_return.reload.status).to eq("intake_needs_assignment")
+          expect(flash[:notice]).to match "Updated status"
+        end
+      end
+
+      context "when the status is the same as the current status" do
+        let(:status) { "intake_in_progress" }
+
+        it "does not create a system status change note" do
+          expect(SystemNote).not_to receive(:create_status_change_note).with(user, tax_return)
```

Comment:
I'd rather see a test that the model's count is unchanged, but this is legit fine.